include/openssl/pkcs7.h

```diff
@@ -272,53 +364,44 @@ OPENSSL_EXPORT int PKCS7_type_is_signed(const PKCS7 *p7);
 // signedAndEnveloped
 OPENSSL_EXPORT int PKCS7_type_is_signedAndEnveloped(const PKCS7 *p7);
 
-
-// PKCS7_sign [Deprecated]
+// Deprecated flags
 //
-// Only |PKCS7_DETACHED| and a combination of
-// "PKCS7_DETACHED|PKCS7_BINARY|PKCS7_NOATTR|PKCS7_PARTIAL" is supported.
-// See |PKCS7_sign| for more details.
+// Not all defined flags are acted upon, and the behavior associated with some
+// flags is performed unconditionally. See each |PKCS7_*| for details.
 
 // PKCS7_DETACHED indicates that the PKCS#7 file specifies its data externally.
 #define PKCS7_DETACHED 0x40
 
 // PKCS7_BINARY disables the default translation to MIME canonical format (as
-// required by the S/MIME specifications).
-// Must be used as "PKCS7_DETACHED|PKCS7_BINARY|PKCS7_NOATTR|PKCS7_PARTIAL".
+// required by the S/MIME specifications). It is assumed in |PKCS7_sign| unless
+// the caller is just bundling certs.
 #define PKCS7_BINARY 0x80
 
-// PKCS7_NOATTR disables usage of authenticatedAttributes.
-// Must be used as "PKCS7_DETACHED|PKCS7_BINARY|PKCS7_NOATTR|PKCS7_PARTIAL".
+// PKCS7_NOATTR disables usage of authenticatedAttributes. It is assumed in
+// |PKCS7_sign| unless the caller is just bundling certs.
 #define PKCS7_NOATTR 0x100
 
-// PKCS7_PARTIAL outputs a partial PKCS7 structure which additional signers and
-// capabilities can be added before finalization.
-// Must be used as "PKCS7_DETACHED|PKCS7_BINARY|PKCS7_NOATTR|PKCS7_PARTIAL".
-#define PKCS7_PARTIAL 0x4000
-
-// PKCS7_TEXT prepends MIME headers for type text/plain to the data. Using this
-// will fail |PKCS7_sign|.
-#define PKCS7_TEXT 0x1
-
 // PKCS7_NOCERTS excludes the signer's certificate and the extra certs defined
-// from the PKCS7 structure. Using this will fail |PKCS7_sign|.
+// from the PKCS7 structure. Using this will fail |PKCS7_sign| unless used as
+// described in |PKCS7_sign|'s documentation.
 #define PKCS7_NOCERTS 0x2
 
-// PKCS7_NOSMIMECAP omits SMIMECapabilities. Using this will fail |PKCS7_sign|.
-#define PKCS7_NOSMIMECAP 0x200
-
-// PKCS7_STREAM returns a PKCS7 structure just initialized to perform the
-// signing operation. Signing is not performed yet. Using this will fail
-// |PKCS7_sign|.
-#define PKCS7_STREAM 0x1000
+// PKCS7_NOVERIFY will skip trust chain verification against the trust store.
+// It will still verify signatures against signer infos included in the PKCS7.
+#define PKCS7_NOVERIFY 0x20
 
-// The following flags are used with |PKCS7_verify| (not implemented).
+// The following flags are used in OpenSSL, but are ignored by AWS-LC. They are
+// defined here solely for build compatibility.
+#define PKCS7_TEXT 0x1
 #define PKCS7_NOSIGS 0x4
 #define PKCS7_NOCHAIN 0x8
 #define PKCS7_NOINTERN 0x10
```

Comment:
`PKCS7_NOINTERN` should be moved out of this list, and documented above. Are there any other places we mention the `PKCS7_NOINTERN` flag?

Reply:
right, fixed.

> Are there any other places we mention the PKCS7_NOINTERN flag?

no.

```
$ ag PKCS7_NOINTERN
crypto/pkcs7/pkcs7.c
1458:    if (!(flags & PKCS7_NOINTERN) && !signer) {

crypto/pkcs7/pkcs7_test.cc
1949:  // attached, but specify |PKCS7_NOINTERN| to ignore bundled certs. this should
1957:                            /*outdata*/ nullptr, /*flags*/ PKCS7_NOINTERN));
1960:                           /*outdata*/ nullptr, /*flags*/ PKCS7_NOINTERN));

include/openssl/pkcs7.h
380:// PKCS7_NOINTERN disables verification against certificate public keys included
383:#define PKCS7_NOINTERN 0x10
453:// If |PKCS7_NOINTERN| is set, this function will not verify against certificate
```